src/clj/puppetlabs/puppetserver/shell_utils.clj

```diff
@@ -39,9 +39,11 @@
 
 (schema/defn ^:always-validate java-exe-options :- ShellUtils$ExecutionOptions
   [{:keys [env in]} :- ExecutionOptions]
-  (doto (ShellUtils$ExecutionOptions.)
-    (.setEnv (if env (ks/mapkeys name env) {}))
-    (.setStdin in)))
+  (let [exe-options (ShellUtils$ExecutionOptions.)]
+    (.setStdin exe-options in)
+    (when env
+      (.setEnv exe-options (ks/mapkeys name env)))
```

Comment:
> Ruby is different because we are running from JRuby and we're intentionally filtering environment variables for the JRubies.

That's not what my point was.  I was interested in knowing if MRI Ruby Puppet does any of its own filtering - or has filtering done on its behalf - of environment variables for a process that it launches via its `ExecutionStub`.  I agree we haven't had any reports about the current environment variable behaviors with Puppet Server's current `ExecutionStub`, though, so I'm good with not looking into this further.